frank-flow/src/frontend/src/app/shared/components/add-dialog/add-dialog.component.ts

```diff
@@ -29,60 +29,65 @@ export class AddDialogComponent {
   }
 
   add(): void {
-    if (this.fileAlreadyExistsInFolder()) {
+    if (!this.currentDirectory.configuration) {
       this.toastr.error(
-        `The file ${this.fileName} already exists in this directory.`,
-        'Error creating'
+        `Please select a configuration or directory.`,
+        `Error creating ${this.isFolder ? 'folder' : 'file'}`
       );
-    } else if (this.currentDirectory.configuration) {
-      this.fileService
-        .updateFileForConfiguration(
-          this.currentDirectory.configuration,
-          this.currentDirectory.path + '/' + this.fileName,
-          this.helloWorldFileTemplate(this.fileName)
-        )
-        .then((response) => {
-          if (response) {
-            this.toastr.success(
-              `The file ${this.fileName} has been created.`,
-              'File created!'
-            );
-          } else {
-            this.toastr.error(
-              `The file ${this.fileName} couldn't be created.`,
-              'Error creating'
-            );
-          }
-        });
+      return;
+    }
+
+    this.createFileOrFolder().then((response) => {
+      this.giveMessage(response);
       this.clearForm();
+      this.clearDirectory();
       this.fileService.fetchFiles();
       this.ngxSmartModalService.close('addDialog');
-    }
+    });
   }
 
-  fileAlreadyExistsInFolder(): boolean {
-    const fileName = this.fileName;
-    const rootDir = this.fileService.configurationFiles.value;
-    const currentDirectory = this.findCurrentDirectory(
-      rootDir[0].content,
-      this.currentDirectory.path
-    );
-    return this.findDuplicateFileName(fileName, currentDirectory);
+  createFileOrFolder(): Promise<Response> {
+    if (this.isFolder) {
+      return this.fileService.createDirectoryForConfiguration(
+        this.currentDirectory.configuration,
+        this.currentDirectory.path + '/' + this.fileName
+      );
+    } else {
+      return this.fileService.createFileForConfiguration(
+        this.currentDirectory.configuration,
+        this.currentDirectory.path + '/' + this.fileName,
+        this.helloWorldFileTemplate(this.fileName)
+      );
+    }
   }
 
-  findCurrentDirectory(directory: any, path: string | undefined): any {
-    if (path === '' || path === undefined) {
-      return directory;
+  giveMessage(response: Response): void {
```

Comment:
if else can be split up into multiple functions.

Reply:
GiveMessage {
GiveSuccessMessage
GiveAlreadyExistsMessage
GiveErrorMessage
}

I dont think that makes it any better

Reply:
It makes the message smaller and according to SOLID it should be split up because the function does multiple things.

Reply:
The function doesn't do multiple things, it gives one off the possible messages